dali/operators/random/rng_base_gpu.cuh

```diff
@@ -28,33 +28,82 @@ namespace dali {
 
 namespace {
 
+template <bool value>
+using bool_const = std::integral_constant<bool, value>;
+
 template <typename T, typename Dist>
-__device__ __inline__ void Generate(BlockDesc<true> desc,
+__device__ __inline__ void Generate(const SampleDesc &sample,
```

Comment:
1. is_noise_gen=True, is_per_channel=True.
Noise generator (applies random noise to an input) that treats channels independently (each channel